File: Tim/Data/CommandData.java
package Tim.Data;

import org.pircbotx.hooks.types.GenericChannelEvent;
import org.pircbotx.hooks.types.GenericMessageEvent;
import org.pircbotx.hooks.types.GenericUserEvent;

public class CommandData<T extends GenericChannelEvent & GenericUserEvent & GenericMessageEvent> {
	public CommandType type      = CommandType.UNKNOWN;
	public String      issuer    = "";
	public String      command   = "";
	public String[]    args      = {};
	public String      argString = "";
	public T           event;

	public void setEvent(T event) {
		this.event = event;
	}

	public GenericMessageEvent getMessageEvent() {
		return event;
	}

	public GenericUserEvent getUserEvent() {
		return event;
	}

	public GenericChannelEvent getChannelEvent() {
		return event;
	}

	public String toString() {
		return String.format("Command: %s, Issuer: %s, argString: %s.", command, issuer, argString);
	}

	public enum CommandType {
		TIMMY_USER, TIMMY_ADMIN, SKYNET_USER, SKYNET_ADMIN, UNKNOWN
	}
}

File: timmy/data/command_data.py
from typing import Optional

from timmy.data.command_type import CommandType
from timmy.data.userdata import UserData


class CommandData:
    type: CommandType
    issuer_data: Optional[UserData]
    issuer: str
    channel: str
    command: str
    args: list
    arg_count: int
    arg_string: str
    prefix: str
    in_pm: bool
    automatic: bool

    def __init__(self):
        self.type = CommandType.UNKNOWN
        self.issuer = ""
        self.channel = ""
        self.command = ""
        self.args = []
        self.arg_count = 0
        self.arg_string = ""
        self.prefix = ""
        self.in_pm = False
        self.automatic = False

    def __str__(self):
        return "Command: %s, Issuer: %s, Args: %s".format(self.prefix + self.command, self.issuer, self.arg_string)
